### ISSUE TITLE
[cython] compatibility with ipython / jedi completion

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#599]
Currently doing tab completing in IPython, I get something like:

```
In [2]: s = GeoSeries([Point(i, i) for i in range(5)])

In [5]: s.<TAB>

I am densified (5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
Invalid argument (must be a Polygon)
Invalid argument (must be a Polygon)
I am densified (external_values, 5 elements)
I am densified (5 elements)
Invalid argument (must be a Polygon)
Invalid argument (must be a Polygon)
^[[56;1R^[[56;1R^[[56;1R^[[56;1R^[[56;1RI am densified (5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
Invalid argument (must be a Polygon)
I am densified (5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
I am densified (5 elements)
```

So it seems to actually execute some of the attributes, which we should somehow prevent (eg overwriting in GeoPandas to not do anythin like GeoSeries.flags). Apart from that, the `_dir_additions` and `_dir_deletions` methods from pandas also invoke two densifications when checking the `.dt` accessor.

Possibly related https://github.com/davidhalter/jedi/issues/931

**Comment by jorisvandenbossche** [comment]
Given that the issue on jedi I linked, had another related issue link which was closed in the meantime, and the jedi changelog is mentioning "Avoiding side effects in REPL completion." (https://github.com/davidhalter/jedi/blob/master/CHANGELOG.rst#0110-2017-09-20), I updated my jedi installation, and most problems are already solved:

```
In [4]: s = GeoSeries([Point(i, i) for i in range(5)])

In [5]: s.<TAB>
I am densified (external_values, 5 elements)
I am densified (external_values, 5 elements)
```

So only the ones from `_dir_additions` and `_dir_deletions` are still present. 
And that's something that could easily be solved. So this seems to be better than first thought!
